### PR TITLE
Send eos to appwriter on failure

### PR DESCRIPTION
**Pull request by frostbyte73**
Fixes track recordings of screen shares (and is an alternative fix for track disconnection)

Will clean up queue/multiqueue related comments in a separate PR

### CODE REVIEW

**Comment by davidzhao** [comment]
which issue does this resolve?

File: pkg/pipeline/input/sdk/appwriter.go [hunk 1]
```diff
@@ -192,29 +192,26 @@ func (w *appWriter) start() {
 				return
 			}
 
-			// when a track is killed, one 8x8 frame is sent by the SFU and needs to be ignored
-			if pkt != nil {
```
Comment (davidzhao) [inline pkg/pipeline/input/sdk/appwriter.go:196]: we no longer need this?

Comment (frostbyte73) [reply pkg/pipeline/input/sdk/appwriter.go:196]: Nope. This was an attempt to get around the final packet and worked for killing a video track, but doesn't work for screen share. The screen share fix of allowing the error and sending an eos to that appwriter also works on regular video tracks

**Comment by frostbyte73** [comment]
> which issue does this resolve?

Updated description

**Review by davidzhao — APPROVED** [review]
nice!

**Pull request merged**